app/backend/src/clj/teet/contract/contract_commands.clj

```diff
@@ -61,6 +64,43 @@
      (merge tempids
             {:company-contract-id new-company-contract-id})))
 
+(defcommand :thk.contract/save-contract-partner-company
+  {:doc "Save a new contract partner"
+   :payload {form-data :form-data
+             contract :contract}
+   :context {:keys [user db]}
+   :project-id nil
+   :authorization {:contracts/contract-editing {}}
+   :pre [^{:error :business-registry-code-in-use}
+         (company-db/business-registry-code-unique?
+           db
+           (company-model/company-business-registry-id-with-country-code form-data))]}
+  (let [company-fields (-> (select-keys
+                             form-data [:db/id
+                                        :company/country
+                                        :company/business-registry-code
+                                        :company/name
+                                        :teet/id])
+                         (assoc :company/emails (str/join "" (:company/emails form-data)))
+                         (assoc :company/phone-numbers (str/join "" (:company/phone-numbers form-data))))
+        company-id (:db/id form-data)
+        contract-eid (:db/id contract)
+        company-contract-uuid (contract-db/contract-partner-relation-entity-uuid db company-id contract-eid)
+        lead-partner? (:company-contract/lead-partner? form-data)
+        tempids
+        (:tempids (tx [(list 'teet.contract.contract-tx/update-contract-partner
+                         contract-eid
+                         lead-partner?
+                         company-id
+                         [(merge
+                            company-fields
+                            (meta-model/modification-meta user))
+                          (merge
+                            {:teet/id company-contract-uuid
+                             :company-contract/lead-partner? (if (true? lead-partner?) true false)}
```

Comment:
We should retract the lead-partner? value all together so it's a bit easier to query for it with the `missing?` function similar to the `:meta/deleted?`

Reply:
and similar to when we add a new lead partner.